polyphemus/lib/etls/redcap/lib/client.rb

```diff
@@ -54,52 +54,30 @@ def initialize eavs, flat_record
       end
 
       def record
-        if @eavs.first[:redcap_repeat_instance]
-          records = @eavs.group_by do |eav|
-            eav[:redcap_repeat_instance].to_s
-          end.map do |repeat_id, record_eavs|
-            next if empty?(repeat_id)
-            make_record(record_eavs, false)
-          end.compact
-          records.empty? ? nil : records
-        else
-          make_record(@eavs)
-        end
+        make_record(@eavs)
       end
 
       private
 
       def empty?(value)
-        value.nil? || value == ''
-      end
-
-      def add_eav?(eav)
-        eav[:value] &&
-          eav[:value] != '' &&
-          ![ 'Not Available', 'Not Reported', 'Not reported' ].include?(eav[:value])
+        value.nil? || value.empty?
       end
 
       def field_name(eav)
         eav[:field_name].to_s.gsub(/_?+[0-9]+$/,'').to_sym
       end
 
-      def make_record(record_eavs, use_flat=true)
+      def make_record(record_eavs)
         record = record_eavs.reduce({}) do |rec,eav|
-          if add_eav?(eav)
-            rec[eav[:field_name].to_sym] = best_value(
-              eav, use_flat
-            )
+          if !empty?(eav[:value])
```

Comment:
A little curious of this logic.  We check if the eav value is not empty, but then we only use it if the inner @flat_record's entry is also empty.  If @flat_record[eav[:field_name]] is not empty, but eav[:value] is, do we intend to ignore it?

Reply:
I think it would not be possible for @flat_record[ eav[:field_name] ] to be present while eav[:value] is absent, although I am a little uncertain. This whole thing is a kludge to account for the fact that the redcap API is broken and does not translate ontology values for `eav` mode (e.g., if you are using the OMIM ontology, it will return the OMIM numeric code in `eav` mode and the OMIM term string in `flat` mode). Therefore, we need to get the record in eav format and in flat format to have the required information. The intent here is to conservatively paste the translated value in the flat record over the untranslated eav value. However, whenever one exists, the other probably ought to as well.

Reply:
Got it.  Hmm, yeah overall the redcap api seems confusing, and worse, not open source.  Who, picks this kind of software? 🤷